src/modules/dpos/validator/components/delegateProfile/delegateVoteButton.js

```diff
@@ -5,24 +5,24 @@ import DialogLink from 'src/theme/dialog/link';
 import { PrimaryButton, SecondaryButton } from 'src/theme/buttons';
 import { useSentVotes } from '../../hooks/queries';
 
-function DelegateVoteButton({ currentAddress, address, isBanned }) {
+function DelegateVoteButton({ address, isBanned, currentAddress }) {
   const { t } = useTranslation();
   const { data: sentVotes, isLoading: sentVotesLoading } = useSentVotes({
-    config: { params: { address: currentAddress } },
+    config: { params: { address } },
   });
 
   const hasSentVoteToDelegate = useMemo(() => {
     if (!sentVotes?.data) return false;
 
-    return sentVotes.data.votes.some(({ delegateAddress }) => delegateAddress === address);
+    return sentVotes.data?.votes?.some?.(({ delegateAddress }) => delegateAddress === address);
   }, [sentVotes]);
 
   return (
     <DialogLink component="editVote">
       {hasSentVoteToDelegate ? (
-        <SecondaryButton disabled={sentVotesLoading || isBanned}>{t('Edit vote')}</SecondaryButton>
+        <SecondaryButton disabled={sentVotesLoading || isBanned || !currentAddress}>{t('Edit vote')}</SecondaryButton>
       ) : (
-        <PrimaryButton disabled={sentVotesLoading || isBanned}>{t('Vote delegate')}</PrimaryButton>
+        <PrimaryButton disabled={sentVotesLoading || isBanned || !currentAddress}>{t('Vote delegate')}</PrimaryButton>
```

Comment:
I don't get this. Why did you add `!currentAddress`?

Reply:
this was so that when an account isn't selected, the `Vote delegate` button is disabled